src/Module/Xrd.php

```diff
@@ -65,13 +65,20 @@ protected function rawContent(array $request = [])
 
 		if (substr($uri, 0, 4) === 'http') {
 			$name = ltrim(basename($uri), '~');
+			$host = parse_url($uri, PHP_URL_HOST);
 		} else {
 			$local = str_replace('acct:', '', $uri);
 			if (substr($local, 0, 2) == '//') {
 				$local = substr($local, 2);
 			}
 
 			$name = substr($local, 0, strpos($local, '@'));
+			$host = ltrim(substr($local, strpos($local, '@')), '@');
+		}
+
+		if (!empty($host) && $host !== DI::baseUrl()->getHost()) {
+			DI::logger()->debug('Invalid host name for xrd query',['host' => $host, 'uri' => $uri]);
```

Comment:
I wouldn't use `debug` here. `debug` should be used for logging regular everyday stuff. But this here is different. So from my perspective `notice` would be best.